libs/i18n/src/lib/services/root-i18n/root-i18n.service.ts

```diff
@@ -40,24 +40,56 @@ export class NgxI18nRootService {
 	 * @param language - The provided language
 	 */
 	public setCurrentLanguage(language: string): void {
-		// Iben: If a language is set that's not part of the available languages, we return a warn
+		// Iben: Save the new language
+		let newLanguage = language;
+
+		// Iben: Check if the new language is part of the available languages
 		if (!this.configuration.availableLanguages.includes(language)) {
+			// Iben: If a language is set that's not part of the available languages, we return a warn
 			console.warn(
 				`NgxI18n: A language, ${language}, was attempted to be set that was not part of the available languages (${this.configuration.availableLanguages.join(
 					', '
 				)})`
 			);
 
-			// Iben: Early exit
-			return;
+			// Iben: If there is already a language set, we early exit and keep the remaining language
+			if (this.currentLanguage) {
+				return;
+			}
+
+			// Iben: If no language exists, we use the default language
+			newLanguage = this.configuration.defaultLanguage;
 		}
```

Comment:
I would move the functionality of this `if (!this.configuration.availableLanguages.includes(language)) { ... }` to a seperate private method `getNewLanguage` or something to isolate this functionality from setting it.

Reply:
Good call!